packages/core/src/browser/source-tree/source-tree.ts

```diff
@@ -15,25 +15,54 @@
  ********************************************************************************/
 
 import { injectable } from 'inversify';
+import { Disposable } from '../../common/disposable';
 import { MaybePromise } from '../../common/types';
-import { TreeImpl, CompositeTreeNode, TreeNode, SelectableTreeNode, ExpandableTreeNode } from '../tree';
-import { TreeElement, CompositeTreeElement, TreeSource } from './tree-source';
+import { CompositeTreeNode, ExpandableTreeNode, SelectableTreeNode, TreeImpl, TreeNode } from '../tree';
+import { CompositeTreeElement, TreeElement, TreeSource } from './tree-source';
 
 @injectable()
 export class SourceTree extends TreeImpl {
 
+    private readonly expanded = new Map<string, number>();
```

Comment:
I don't think changing core source tree should be required to fix something in the debug extension. There should be cleaner way just in the debug extension.

I was thinking to write a node program which sleeps for several seconds and put a breakpoint before and after to test how VS Code behaves in this case. It could be that they simply does not discard state from previous pause or even keep showing it to a user. And then align our behaviour with that. It probably will require to reconsider how we store threads, right now we discard them on continue.